backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java

```diff
@@ -0,0 +1,666 @@
+    if (workspaceId != null && !workspaceId.isEmpty()) {
+      switch (workspaceType) {
+        case ORGANIZATION:
+          if (datasetVisibility.equals(DatasetVisibilityEnum.DatasetVisibility.ORG_SCOPED_PUBLIC)) {
+            String orgDatasetReadRoleBindingName =
+                roleService.buildRoleBindingName(
+                    ModelDBConstants.ROLE_DATASET_READ_ONLY,
+                    datasetId,
+                    new CollaboratorOrg(workspaceId),
+                    ModelResourceEnum.ModelDBServiceResourceTypes.DATASET.name());
+            if (orgDatasetReadRoleBindingName != null && !orgDatasetReadRoleBindingName.isEmpty()) {
+              workspaceRoleBindings.add(orgDatasetReadRoleBindingName);
+            }
+          }
+          break;
+        case USER:
+        default:
+          break;
+      }
+    }
+    List<String> orgWorkspaceRoleBindings =
+        roleService.getWorkspaceRoleBindings(
+            workspaceId,
+            workspaceType,
+            datasetId,
+            ModelDBConstants.ROLE_DATASET_ADMIN,
+            ModelResourceEnum.ModelDBServiceResourceTypes.DATASET,
+            datasetVisibility.equals(DatasetVisibilityEnum.DatasetVisibility.ORG_SCOPED_PUBLIC),
+            DATASET_GLOBAL_SHARING);
+
+    if (orgWorkspaceRoleBindings != null && !orgWorkspaceRoleBindings.isEmpty()) {
+      workspaceRoleBindings.addAll(orgWorkspaceRoleBindings);
+    }
+    return workspaceRoleBindings;
+  }
+
+  private void deleteDatasetVersions(Session session) {
+    LOGGER.debug("DatasetVersion deleting");
+    String alias = "dv";
+    String deleteDatasetVersionsQueryString =
+        new StringBuilder("FROM ")
+            .append(DatasetVersionEntity.class.getSimpleName())
+            .append(" ")
+            .append(alias)
+            .append(" WHERE ")
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.DELETED)
+            .append(" = :deleted ")
+            .toString();
+    Query datasetVersionDeleteQuery = session.createQuery(deleteDatasetVersionsQueryString);
+    datasetVersionDeleteQuery.setParameter("deleted", true);
+    LOGGER.debug("DatasetVersion delete query: {}", datasetVersionDeleteQuery.getQueryString());
+    List<DatasetVersionEntity> datasetVersionEntities = datasetVersionDeleteQuery.list();
+
+    // Remove roleBindings by accessible datasetVersions
+    List<String> roleBindingNames = new LinkedList<>();
+    for (DatasetVersionEntity datasetVersionEntity : datasetVersionEntities) {
+      String ownerRoleBindingName =
+          roleService.buildRoleBindingName(
+              ModelDBConstants.ROLE_DATASET_VERSION_OWNER,
+              datasetVersionEntity.getId(),
+              datasetVersionEntity.getOwner(),
+              ModelResourceEnum.ModelDBServiceResourceTypes.DATASET_VERSION.name());
+      if (ownerRoleBindingName != null && !ownerRoleBindingName.isEmpty()) {
+        roleBindingNames.add(ownerRoleBindingName);
+      }
```

Comment:
Same.

Reply:
Created separate functions called `deleteRoleBindingsFor*` and moved all delete role bindings code in that function for each entity.